### PR TITLE
Add the custom UrlRelative policy

### LINKED ISSUE #47: Custom UrlRelative policy
It might be useful to have a `UrlRelative::Evaluate(Box<Fn(String) -> Option<String>>)` (or `FnMut`), where the function gets passed the previous URL and returns a new one. This allows the caller to do custom sanitization or rewriting of the URLs without going through `to_dom_node`.

Do we want to implement something like this?

### PR DESCRIPTION
**Pull request by notriddle**
Fixes #47

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -816,6 +818,28 @@ impl<'a> Builder<'a> {
                         attr.value = format_tendril!("{}", url);
                     }
                 }
+            } else if let UrlRelative::Evaluate(ref evaluate) = self.url_relative {
+                let mut drop_attrs = Vec::new();
+                let mut attrs = attrs.borrow_mut();
+                for (i, attr) in attrs.iter_mut().enumerate() {
+                    if is_url_attr(&*name.local, &*attr.name.local) {
+                        let new_value = evaluate.evaluate(&*attr.value)
+                            .map(|new| format_tendril!("{}", new));
```
Comment (lnicola) [inline src/lib.rs]: Maybe `.map(Tendril::from_str)` or something like that.

File: src/lib.rs [hunk 2]
```diff
@@ -880,20 +904,63 @@ fn is_url_attr(element: &str, attr: &str) -> bool {
...
+///         let a = ammonia::Builder::new()
+///             .url_relative(ammonia::UrlRelative::Evaluate(Box::new(evaluate)))
+///             .clean("<a href=/test/path>fixed</a><a href=path>passed</a><a href=http://google.com/>skipped</a>")
+///             .to_string();
+///         assert_eq!(a, "<a href=\"/root/test/path\" rel=\"noopener noreferrer\">fixed</a><a href=\"path\" rel=\"noopener noreferrer\">passed</a><a href=\"http://google.com/\" rel=\"noopener noreferrer\">skipped</a>");
+///     }
+///
+/// This function is only applied to relative URLs.
+/// To filter all of the URLs,
+/// use the not-yet-implemented Content Security Policy.
```
Comment (lnicola) [inline src/lib.rs]: I think we should implement `Debug` manually (`C-DEBUG`).

**Comment by lnicola** [comment]
Bikeshed: would `Custom` be a better name for this?

I think it looks OK with the exception of that `Debug` impl.

**Comment by notriddle** [comment]
bors r= @lnicola

**Pull request merged**